pkg/handler/repositories.go

```diff
@@ -131,12 +140,19 @@ func (rh *RepositoryHandler) bulkCreateRepositories(c echo.Context) error {
 		newRepositories[i].FillDefaults()
 	}
 
-	response, err := rh.RepositoryDao.BulkCreate(newRepositories)
-	if err != nil {
-		return c.JSON(httpCodeForError(err), response)
+	responses, errs := rh.RepositoryDao.BulkCreate(newRepositories)
+	if len(errs) > 0 {
+		errorResponses := make([]api.ErrorResponse, len(errs))
+		for i := 0; i < len(errs); i++ {
+			if errs[i] != nil {
+				errorResponses[i] = api.NewErrorResponse(httpCodeForError(errs[i]), "Error creating repository", errs[i].Error())
+			} else {
+				errorResponses[i] = api.ErrorResponse{}
+			}
+		}
+		return c.JSON(http.StatusOK, echo.Map{"errors": errorResponses})
```

Comment:
This I made a TODO after I opened the PR because I thought of it after. It will return the most generally applicable error, either 400 or 500, depending on what the errors are.